modules/Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections/deploy.bicep

```diff
@@ -0,0 +1,86 @@
+@sys.description('Conditional. The name of the parent network manager. Required if the template is used in a standalone deployment.')
+param networkManagerName string
+
+@sys.description('Conditional. The name of the parent security admin configuration. Required if the template is used in a standalone deployment.')
+param securityAdminConfigurationName string
+
+@maxLength(64)
+@sys.description('Required. The name of the admin rule collection.')
+param name string
+
+@maxLength(500)
+@sys.description('Optional. A description of the admin rule collection.')
+param description string = ''
+
+@sys.description('Required. List of network groups for configuration. An admin rule collection must be associated to at least one network group.')
+param appliesToGroups array
+
+@sys.description('Optional. List of rules for the admin rules collection. Security admin rules allows enforcing security policy criteria that matches the conditions set. Warning: A rule collection without rule will cause a deployment configuration for security admin goal state in network manager to fail.')
+param rules array
+
+@sys.description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableDefaultTelemetry bool = true
+
+var enableReferencedModulesTelemetry = false
+
+resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (enableDefaultTelemetry) {
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource networkManager 'Microsoft.Network/networkManagers@2022-07-01' existing = {
+  name: networkManagerName
+
+  resource securityAdminConfiguration 'securityAdminConfigurations@2022-07-01' existing = {
+    name: securityAdminConfigurationName
+  }
+}
+
+resource ruleCollection 'Microsoft.Network/networkManagers/securityAdminConfigurations/ruleCollections@2022-07-01' = {
+  name: name
+  parent: networkManager::securityAdminConfiguration
+  properties: {
+    description: description
+    appliesToGroups: appliesToGroups
+  }
+}
+
+module securityAdminConfigurations_rules 'rules/deploy.bicep' = [for (rule, index) in rules: {
+  name: '${uniqueString(deployment().name)}-RuleCollections-Rules-${index}'
+  params: {
+    networkManagerName: networkManager.name
+    securityAdminConfigurationName: securityAdminConfigurationName
+    ruleCollectionName: ruleCollection.name
+    name: rule.name
+    access: rule.access
+    description: contains(rule, 'description') ? rule.description : ''
+    destinationPortRanges: contains(rule, 'destinationPortRanges') ? rule.destinationPortRanges : []
+    destinations: contains(rule, 'destinations') ? rule.destinations : []
```

Comment:
@AlexanderSehr I have added the following new properties `destinations` and `sources` to handle a use case where multiple IPs are required for either a source or a destination. 

**BEFORE**
```bicep
destinations: !empty(destinationsAddressPrefix) && !empty(destinationsAddressPrefixType) ? [
      {
        addressPrefix: destinationsAddressPrefix // this value is defined as a parameter
        addressPrefixType: destinationsAddressPrefixType // this value is defined as a parameter
      }
    ] : [ ] 
```

**AFTER**

```bicep
    destinations: !empty(destinationsAddressPrefix) && !empty(destinationsAddressPrefixType) && empty(destinations) ? [ // added another condition
      {
        addressPrefix: destinationsAddressPrefix
        addressPrefixType: destinationsAddressPrefixType
      }
    ] : destinations // this is no longer passed empty here
```
-------------

Now users have the choice to use a single entry using the parameters `destinationsAddressPrefix` and `destinationsAddressPrefixType` (Method-1) 

**OR**

provide the input into the `destinations` object which can support multiple entries (Method-2):

![image](https://user-images.githubusercontent.com/28486158/210691601-1ccf93df-399b-4068-a586-51a9738d3c34.png)
In this example: **Method-2** takes precedence over **Method-1**

I wanted to highlight this change as I've introduced it after the review. Let me know your thoughts on this one. Should we:
- Allow both methods for flexibility OR
- get rid of **method-1** and only support **method-2**

Method-2 is how we always do things for child properties. But I thought of introducing some flexibility.

Reply:
Hey @ahmadabdalla, while I like the flexibility, I wonder if it would ultimately lead to more confusion as one has to understand two approaches instead of one - and use either one based on the use case.
To this end I'm inclined to vote for option 2, that is, `get rid of method-1 and only support method-2`. But I'll leave it up to you :D

Reply:
Thanks @AlexanderSehr After giving it some thought, I've decided to stick to `method-2` and remove support for `method-1`. Will be resolving this comment as well. Based on the outcome of this discussion.